### PR TITLE
feat: refactored Instance ImportTask

### LINKED ISSUE #306: Pack icon doesn't export with the pack itself
### Operating System

_No response_

### Version of Prism Launcher

Prism Launcher 5.0

### Version of Qt

Qt 6.4.0

### Description of bug

When exporting a pack the icon is not exported along with the rest of the pack itself.  This results in a pack that won't have an icon when imported even though it is defined.

Secondary bug: If you import a zip that has an icon defined but isn't in the zip itself and that icon already exists in the icons folder it will be deleted instead of re-used.  This results in all other instances that happened to use that icon to revert to a default icon instead.

### Steps to reproduce

Import or create a pack and set an icon.  The effect is more apparent if you set a custom icon since it won't be in the default set.

Workaround: You can copy the set icon to the root of the instance folder (along with the mmc-pack.json file) and it will be saved into the exported package.  When imported the file gets correctly copied and used.

### Suspected cause

In MultiMC the icon for the pack was always included in the export zip.  I'm thinking this got lost when the packaging format was changed?

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

### LINKED ISSUE #1629: Could not rename the overrides folder: overrides when installing a modpack
### Operating System

Windows

### Version of Prism Launcher

7.2

### Version of Qt

6.5.1

### Description of bug

Whenever I try to install any modpack with Prism launcher I get the same error over and over, no matter whether I have opened it with admin or not. The error is shown in the screenshot below. On some rare occasions, it has worked and have been able to install a modpack. But that has only been 4 out of nearly 100 attempts. Just to be specific this issue only affects the installation of mod packs from 3rd party sources and importing from files.
image_2023-09-16_155559558

### Steps to reproduce

- Launch Prism launcher 7.2
- Click Add Instance
- import a modpack or install one from a 3rd party source
- wait for the error to appear

### Suspected cause

I only have suspicions on what the issue might be because I'm no specialist at this but I suspect It might possibly be that Prism launcher doesn't have permission to rename files on my computer.

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

### PR DESCRIPTION
**Pull request by Trial97**
Added a sub-task to monitor the extracting progress.
And made sure that we report correctly that we are searching the zip file to determine the mod pack type(previously for a large invalid pack the message would be `Extracting modpack` and noting moved to denote that there is progress).
Now on that step changed the status to `Attempting to determine instance type` and the details of the task are now changing with the name of the file that we look at.

fixes #306
fixes #1629

### CODE REVIEW

**Comment by Scrumplex** [comment]
This is fairly significant. Moving to 9.0

**Review by DioEgizio — APPROVED** [review]

**Review by DioEgizio — APPROVED** [review]

File: launcher/MMCZip.cpp [hunk 1]
```diff
@@ -332,9 +333,20 @@ std::optional<QStringList> extractSubDir(QuaZip* zip, const QString& subdir, con
...
+        auto fileInfo = QFileInfo(target_file_path);
+        if (fileInfo.isFile()) {
+            auto permissions = fileInfo.permissions();
+            auto maxPermisions = QFileDevice::Permission::ReadUser | QFileDevice::Permission::WriteUser | QFileDevice::Permission::ExeUser |
+                                 QFileDevice::Permission::ReadGroup | QFileDevice::Permission::ReadOther;
+            auto minPermisions = QFileDevice::Permission::ReadUser | QFileDevice::Permission::WriteUser;
+
+            auto newPermisions = (permissions & maxPermisions) | minPermisions;
+            if (newPermisions != permissions) {
+                if (!QFile::setPermissions(target_file_path, permissions)) {
```
Comment (TheKodeToad) [inline launcher/MMCZip.cpp]: Unless I'm mistaken, this seems to not be changing the file's permissions, as it is setting to what they were to begin with

**Review by TheKodeToad — COMMENTED** [review]
Mostly looks good

File: launcher/MMCZip.cpp (hunk 1 above)
Comment (Trial97) [reply launcher/MMCZip.cpp]: You are right regarding that I fixed it so give me the approval, please.

**Review by TheKodeToad — APPROVED** [review]

**Pull request merged**